Northstar.CustomServers/mod/scripts/vscripts/sh_utility_all.gnut

```diff
@@ -1540,6 +1540,20 @@ array<string> function GetAvailableTitanRefs( entity player )
 	return availableTitanRefs
 }
 
+void function RecalculateHighestTitanFDLevel( entity player )
+{
+	int enumCount = PersistenceGetEnumCount( "titanClasses" )
+	int highestAegis = 0
+	for ( int i = 0; i < enumCount; i++ )
+	{
+		string enumName = PersistenceGetEnumItemNameForIndex( "titanClasses", i )
+		int aegisLevel = FD_TitanGetLevelForXP( enumName, FD_TitanGetXP( player, enumName ) )
+		if ( highestAegis < aegisLevel )
+			highestAegis = aegisLevel
+	}
+	player.SetPersistentVar( "fdStats.highestTitanFDLevel", highestAegis )
```

Comment:
The fact that Respawn stores highest Aegis rank in a separate var instead of calculating it on the fly when needed is kinda stupid ngl
(this is not a review on the code, just a rant)